### PR TITLE
Various sqlite backend optimizations and cleanups

### PR DESCRIPTION
**Pull request by pmatilai**

### CODE REVIEW

File: lib/backend/sqlite.c [hunk 1]
```diff
@@ -584,11 +558,8 @@ static rpmRC sqlite_idxdbIter(dbiIndex dbi, dbiCursor dbc, const char *keyp, siz
 	    dbc->key = sqlite3_column_blob(dbc->stmt, 0);
 	}
 	dbc->keylen = sqlite3_column_bytes(dbc->stmt, 0);
-	if (set) {
-	    dbiCursor kc = dbiCursorInit(dbi, 0);
-	    rc = sqlite_idxdbByKey(dbi, kc, dbc->key, dbc->keylen, set);
-	    dbiCursorFree(dbi, kc);
-	}
+	if (set)
+	    rc = sqlite_idxdbByKey(dbi, dbc->subc, dbc->key, dbc->keylen, set);
```
Comment (Conan-Kudo) [inline lib/backend/sqlite.c]: I think there's an error here. It looks like there should be braces here...

File: lib/backend/sqlite.c [hunk 2]
```diff
@@ -584,11 +558,8 @@ static rpmRC sqlite_idxdbIter(dbiIndex dbi, dbiCursor dbc, const char *keyp, siz
...
 	rc = RPMRC_OK;
```
Comment (Conan-Kudo) [inline lib/backend/sqlite.c]: Is this statement supposed to be here? It essentially clobbers the result from the line above...

File: lib/backend/sqlite.c (hunk 1 above)
Comment (pmatilai) [reply lib/backend/sqlite.c]: What makes you think that? There's no need for braces for a single line.

File: lib/backend/sqlite.c (hunk 2 above)
Comment (pmatilai) [reply lib/backend/sqlite.c]: It's a bit fishy alright, but it's not being introduced in this patch.

IIRC it's there for a reason though, and with this patch all the more so.

File: lib/backend/sqlite.c (hunk 1 above)
Comment (Conan-Kudo) [reply lib/backend/sqlite.c]: Well, aside from a couple lines up there is braces used even with a single line branch on `if`, the `rc = RPMRC_OK` made me think there's some kind of error here.

Comment (pmatilai) [reply lib/backend/sqlite.c]: The brace for single line items varies a lot throughout rpm, the only hard rule there is that anything spanning multiple lines in the source must use braces, even if the compiler thinks it's just one line.

I generally dislike unnecessary braces but sometimes they are introduced in anticipation for surrounding code, at other times they get left behind from other code changes, at other times they add to readability and at others, they're just ugly.

**Comment by pmatilai** [comment]
With sqlite changes, it's worth remembering it's all very much still a work-in-progress thing and should be reviewed with different mindset than changes to released features and such. The whole db schema is likely to be revamped etc, this is all just trying out things.

**Pull request merged**